### PR TITLE
Update base scenarios for the Mitaka release.

### PR DESCRIPTION
**Pull request by msimonin**
I've updated the two basic scenarios to support Mitaka release.

This PR is about starting the discussion about : 
- The mitaka support 
- The way we reorganize the git to support the future releases of OpenStack

For the second item I propose to 
- create a branch to support the liberty release
- keep master up-to-date with the latest release (here Mitaka)

In the future old release will be supported in dedicated branches and current release will be in master.

### CODE REVIEW

**Comment by msimonin** [comment]
Just rebased on master following this commit : https://github.com/grid5000/xp5k-openstack/commit/0b151e999fe5b2c88eac9b3184f8704a3a2fd6c5

**Comment by msimonin** [comment]
Just rebased on master following this commit : 339f7e27fb310b7e3ff847d14effc0a0dad3b593

File: scenarios/multinodes/puppet/modules/scenario/manifests/openstack/controller.pp [hunk 1]
```diff
@@ -53,8 +54,8 @@
   class { '::neutron::server':
     database_connection => "mysql://neutron:neutron@${controller_public_address}/neutron?charset=utf8",
     auth_password       => $admin_password,
-    identity_uri        => "http://${controller_public_address}:35357/",
-    auth_uri            => "http://${controller_public_address}:5000",
+    auth_uri            => "http://${controler_public_address}:5000",
+    auth_url            => "http://${controler_public_address}:35357",
```
Comment (priteau) [inline scenarios/multinodes/puppet/modules/scenario/manifests/openstack/controller.pp:58]: These variables should remain controller_public_address (with two L).

**Comment by priteau** [comment]
Is it planned to merged this support for Mitaka?

**Comment by msimonin** [comment]
10 months already ! We should ping @pmorillon to get his insights ;)

**Comment by pmorillon** [comment]
Pourquoi ne pas créer un scenario dans un repo différent ? Perso, j'ai pas le temps de maintenir toutes les versions d'Openstack. Xp5k-openstack avait été conçu dans ce but.

**Comment by lnussbaum** [comment]
Hi,

Pierre, Matthieu, can you provide some context about this pull request ?

On the Grid'5000 side, we need to decide whether we want to stick with xp5k-openstack (and if we move to Mitaka, support the move in our tutorials), or if we should move to another deployment engine (e.g. ENOS, which gained popularity recently).

I don't have any definitive idea about this (I'm attending a meeting later this week where deployment tools will be discussed, so I hope to make progress about that).

**Comment by priteau** [comment]
We can now use the Liberty scenario on Chameleon using Ubuntu 14.04. We would like to use Ubuntu 16.04 instead which comes with Mitaka in base repositories (or, even better, the most recent compatible OpenStack release).

**Comment by lnussbaum** [comment]
Pierre, have you looked at ENOS ? See http://enos.readthedocs.io/en/latest/index.html
I wonder if we should just ditch the puppet-based deployment and switch to ENOS. It seems to be the tool of choice of the user community around the DISCOVERY project here.
There's at least basic support for using ENOS on Chameleon. @msimonin can probably say more about it if needed.

**Comment by priteau** [comment]
@lnussbaum I know that @msimonin is working on porting ENOS to Chameleon. But since we have a deployment based on xp5k which works well for Liberty, having it support Mitaka would be great.

**Comment by lnussbaum** [comment]
@priteau we discussed the status of openstack deployment during our last CT meeting. The conclusion is that for now, we will "wait and see", and revisit whether we want to switch to ENOS or update xp5k-openstack during the preparation for the next Grid'5000 school. (our main concern is that, if we start to update xp5k-openstack, we also need to make sure that our documentation still match the tool).

So for now, the best solution for you is probably to maintain the Mitaka version in a fork.

@msimonin FYI I just added a pointer to ENOS in https://www.grid5000.fr/mediawiki/index.php?title=OpenStack_deployment_on_Grid5000&type=revision&diff=65956&oldid=63130 . however, if you get the chance to work on the issues identified in https://www.grid5000.fr/mediawiki/index.php/TechTeam:CT-161#Quel_support_pour_le_d.C3.A9ploiement_d.27OpenStack_.3F_.28Lucas.29 (mainly deployment time and tutorials), that would be great. I'll try to work on the jenkins tests at some point.

**Comment by msimonin** [comment]
@lnussbaum ack.

**Comment by pmorillon** [comment]
I worked on it this morning, there will be several branches, liberty/dev, liberty/stable, mitaka/dev, mitaka/stable...
The starter_kit works now with mitaka recipes. multinodes scenario will be merged next week.

**Pull request closed without merging**